infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/InfinispanRemoteDialect.java

```diff
@@ -0,0 +1,380 @@
+				String fieldName = mapper.convertColumnNameToFieldName( columnNames[i] );
+				if ( firstIteration ) {
+					bqEnd = qb.having( fieldName ).eq( key.getColumnValues()[i] );
+					firstIteration = false;
+				}
+				else {
+					bqEnd = bqEnd.and().having( fieldName ).eq( key.getColumnValues()[i] );
+				}
+			}
+			Query query = bqEnd.toBuilder().build();
+			Map<RowKey,Map<String, Object>> resultsCollector = new HashMap<>();
+			try ( CloseableIterator<Entry<Object,Object>> iterator = c.retrieveEntriesByQuery( query, null, 100 ) ) {
+				while ( iterator.hasNext() ) {
+					Entry<Object,Object> e  = iterator.next();
+					ProtostreamPayload value = ( (ProtostreamPayload) e.getValue() );
+					Map<String, Object> entryObject = value.toMap();
+					RowKey entryKey = value.asRowKey( key );
+					resultsCollector.put( entryKey, entryObject );
+				}
+			}
+			return resultsCollector;
+		} );
+	}
+
+	@Override
+	public Association createAssociation(AssociationKey key, AssociationContext associationContext) {
+		Map<RowKey, Map<String, Object>> associationMap = new HashMap<RowKey, Map<String,Object>>();
+		return new Association( new MapAssociationSnapshot( associationMap ) );
+	}
+
+	@Override
+	public void insertOrUpdateAssociation(AssociationKey key, Association association, AssociationContext associationContext) {
+		final String cacheName = key.getTable();
+		final ProtoStreamMappingAdapter mapper = provider.getDataMapperForCache( cacheName );
+		final boolean embeddedInEntity = associatonTypeIsForeignKeyEmbeddedInEntity( key, association, mapper );
+		if ( embeddedInEntity ) {
+			insertOrUpdateAssociationEmbeddedInEntity( key, association, associationContext, mapper, cacheName );
+		}
+		else {
+			insertOrUpdateAssociationMappedAsDedicatedEntries( key, association, associationContext, mapper, cacheName );
+		}
+	}
+
+	private void insertOrUpdateAssociationMappedAsDedicatedEntries(AssociationKey key, Association association, AssociationContext associationContext,
+			ProtoStreamMappingAdapter mapper, String cacheName) {
+		log.debugf( "insertOrUpdateAssociation for key '%s' on cache '%s', mapped as dedicated entries in ad-hoc table", key, cacheName );
+		final List<AssociationOperation> operations = association.getOperations();
+		for ( AssociationOperation ao : operations ) {
+			AssociationOperationType type = ao.getType();
+			RowKey rowKey = ao.getKey();
+			ProtostreamId idBuffer = mapper.createIdPayload( rowKey.getColumnNames(), rowKey.getColumnValues() );
+			switch ( type ) {
+				case PUT:
+					ProtostreamPayload valuePayloadForPut = mapper.createValuePayload( ao.getValue() );
+					mapper.withinCacheEncodingContext( c -> c.put( idBuffer, valuePayloadForPut ) );
+					break;
+				case REMOVE:
+					mapper.withinCacheEncodingContext( c -> c.remove( idBuffer ) );
+					break;
+				case CLEAR:
+					throw new AssertionFailure( "Request for CLEAR operation on an association mapped to dedicated entries. Makes no sense?" );
```

Comment:
@sanne I think you got the wrong picture. It's not a clear of the association table. It's a clear of all the rowkey for that specific association entry (i.e. the assoc info dedicated to a specific entity).

So it would clear all entries returned by the query done in `getAssociation` (that's virtual of course and you can rely on the snapshot for that list).